searchlib/src/tests/features/internal_max_reduce_prod_join_feature/internal_max_reduce_prod_join_feature_test.cpp

```diff
@@ -26,34 +27,65 @@ struct SetupFixture
 {
     InternalMaxReduceProdJoinBlueprint blueprint;
     IndexEnvironment indexEnv;
-    SetupFixture(const vespalib::string &attrName)
+    SetupFixture()
         : blueprint(),
           indexEnv()
     {
-        FieldInfo attrInfo(FieldType::ATTRIBUTE, CollectionType::ARRAY, attrName, 0);
-        attrInfo.set_data_type(DataType::INT64);
+        addAttribute("double", DataType::DOUBLE);
+        addArrayAttribute("longarray", DataType::INT64);
+        addArrayAttribute("intarray", DataType::INT32);
+        addArrayAttribute("doublearray", DataType::DOUBLE);
+    }
+
+    void addArrayAttribute(const vespalib::string& name, const DataType& dataType) {
+        FieldInfo attrInfo(FieldType::ATTRIBUTE, CollectionType::ARRAY, name, 0);
+        attrInfo.set_data_type(dataType);
+        indexEnv.getFields().push_back(attrInfo);
+    }
+
+    void addAttribute(const vespalib::string& name, const DataType& dataType) {
+        FieldInfo attrInfo(FieldType::ATTRIBUTE, CollectionType::SINGLE, name, 0);
+        attrInfo.set_data_type(dataType);
         indexEnv.getFields().push_back(attrInfo);
     }
 };
 
-TEST_F("require that blueprint can be created", SetupFixture("attr"))
+TEST_F("require that blueprint can be created", SetupFixture())
 {
     EXPECT_TRUE(FTA::assertCreateInstance(f.blueprint, "internalMaxReduceProdJoin"));
 }
 
-TEST_F("require that setup fails if attribute does not exist", SetupFixture("attr"))
+TEST_F("require that setup fails if attribute does not exist", SetupFixture())
 {
     FTA::FT_SETUP_FAIL(f.blueprint, f.indexEnv, StringList().add("foo").add("bar"));
 }
 
-TEST_F("require that setup succeeds with attribute and query parameters", SetupFixture("attr"))
+TEST_F("require that setup fails if attribute is of wrong type", SetupFixture())
```

Comment:
nitpicking, but testing with a single value long attribute would be better; keeping as close to what would work as possible (1 step away instead of 2)